File: src/py_pkg/py_pkg/custom_interface_node.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from custom_interfaces.msg import MinimalInterface


class CustomInterfaceNode(Node):

    def __init__(self):
        super().__init__("custom_interface_node")
        self.publisher_ = self.create_publisher(MinimalInterface, "custom_interface", 10)
        self.timer_ = self.create_timer(0.5, self.publish_example)
        self.get_logger().info("Custom interface node is running ...")

    def publish_example(self):
        msg = MinimalInterface()
        msg.a = 12.56
        msg.b = True
        msg.c = "Basic text"
        self.publisher_.publish(msg)

def main(args=None):
    rclpy.init(args=args)
    node = CustomInterfaceNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()

File: src/cpp_pkg/src/custom_interface_node.cpp
#include "rclcpp/rclcpp.hpp"
#include "custom_interfaces/msg/minimal_interface.hpp"

using namespace std::chrono_literals;

class CustomInterfaceNode : public rclcpp::Node
{
    public:
        CustomInterfaceNode() : Node("custom_interface_node")
        {
            publisher_ = this->create_publisher<custom_interfaces::msg::MinimalInterface>("custom_interface", 10);
            timer_ = this->create_wall_timer(500ms, std::bind(&CustomInterfaceNode::publish_example, this));
            RCLCPP_INFO(this->get_logger(), "Custom interface node is running ...");
        }
        
    private:
        void publish_example()
        {
            auto msg = custom_interfaces::msg::MinimalInterface();
            msg.a = 12.56;
            msg.b = true;
            msg.c = "Basic text";
            publisher_->publish(msg);
        } 
    
        rclcpp::Publisher<custom_interfaces::msg::MinimalInterface>::SharedPtr publisher_;
        rclcpp::TimerBase::SharedPtr timer_;
}; 

int main(int argc, char **argv){

    rclcpp::init(argc, argv);
    auto node = std::make_shared<CustomInterfaceNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}